graphbolt/src/utils.h

```diff
@@ -19,6 +19,13 @@ inline bool is_accessible_from_gpu(torch::Tensor tensor) {
   return tensor.is_pinned() || tensor.device().type() == c10::DeviceType::CUDA;
 }
 
+/**
+ * @brief Checks whether the tensor is stored on the GPU.
+ */
+inline bool is_on_gpu(torch::Tensor tensor) {
+  return tensor.device().type() == c10::DeviceType::CUDA;
```

Comment:
How about this?
```
template<typename... Args>
inline bool is_cuda(const Args&... args) {
    return (... && (args.is_cuda()));
}

```

Reply:
or directly call `nodes.is_cuda()` in user code?

Reply:
I am not sure if there is a `is_cuda()` method.

Reply:
But I will try getting a variable number of argument function working.

Reply:
https://github.com/pytorch/pytorch/blob/4a10e9eed4f8faf3e68106e5e86559158d83a200/aten/src/ATen/core/TensorBase.h#L430 ?

Reply:
I wonder why they have this comment there:
https://github.com/pytorch/pytorch/blob/4a10e9eed4f8faf3e68106e5e86559158d83a200/aten/src/ATen/core/TensorBase.h#L431
```c++
// NB: this is not a native function to avoid dispatching overhead.
```

Reply:
https://github.com/pytorch/pytorch/blob/4a10e9eed4f8faf3e68106e5e86559158d83a200/c10/core/TensorImpl.h#L1097

Reply:
or use this one https://github.com/pytorch/pytorch/blob/4a10e9eed4f8faf3e68106e5e86559158d83a200/c10/core/Device.h#L81?

Reply:
The current code already uses the last one you just pointed out.

Reply:
Oh, you mean calling is_cuda on the device, Ok let me do that as the implementation is exactly the same as ours.

Reply:
I think the guideline is always use available APIs(and c++ APIs are usually consistent with python in pytorch) except for performance or other issue is identified.